pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+		err = createSecretForOpenshiftRoute(comp, svc)
+		if err != nil {
+			svc.Log.Error(err, "Failed to create Collabora Openshift Secret "+err.Error())
+			return runtime.NewWarningResult("Failed to add Collabora Openshift Route: " + err.Error())
+		}
+	}
+
+	svc.Log.Info("Waiting for Nextcloud release to be ready")
+
+	releaseExists := svc.WaitForObservedDependencies(comp.GetName(), comp.GetName()+"-release")
+	if !releaseExists {
+		return runtime.NewWarningResult("Nextcloud instance not yet ready")
+	}
+
+	svc.Log.Info("Nextcloud release is ready, creating STS, Service, Ingress, Issuer, Certificate, ConfigMap, Job")
+
+	// Add Collabora STS
+	err = AddCollaboraSts(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora STS: " + err.Error())
+	}
+
+	// Add Collabora service
+	err = AddCollaboraService(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora Service: " + err.Error())
+	}
+
+	// Add Collabora ingress
+	err = AddCollaboraIngress(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora Ingress: " + err.Error())
+	}
+
+	// Create install Collabora config map
+	err = createInstallCollaboraConfigMap(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora ConfigMap: " + err.Error())
+	}
+
+	// Create install Collabora job
+	err = createInstallCollaboraJob(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora install Job: " + err.Error())
+	}
+
+	return runtime.NewNormalResult("Collabora deployed and configured!")
+}
+
+func AddCollaboraSts(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+
+	sts := &v1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels: map[string]string{
+				"app": comp.GetName() + "-collabora-code",
+			},
+		},
+		Spec: v1.StatefulSetSpec{
+			Replicas: ptr.To[int32](1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": comp.GetName() + "-collabora-code"},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   comp.GetName() + "-collabora-code",
+					Labels: map[string]string{"app": comp.GetName() + "-collabora-code"},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: comp.GetName() + "-collabora-code-sa",
+					// SecurityContext: &corev1.PodSecurityContext{
+					// 	// https://github.com/CollaboraOnline/online/blob/master/docker/from-packages/Dockerfile#L138
+					// 	RunAsUser: ptr.To[int64](100),
+					// },
+					Containers: []corev1.Container{
+						{
+							Env: []corev1.EnvVar{
+								{
+									Name:  "DONT_GEN_SSL_CERT",
+									Value: "true",
+								},
+								{
+									// https://github.com/nextcloud/all-in-one/discussions/2314
+									// https://github.com/CollaboraOnline/online/issues/3102
```

Comment:
I want to leave it here because those are links to github issues explaining why we have such logic implemented, as Collabora is not cloud-native software it requires plenty of hack, to make it running especially in Openshift